tests/sentry/models/test_group.py

```diff
@@ -36,56 +41,68 @@ def test_is_resolved(self):
+        group = self.create_group(project=project)
         assert group.get_latest_event() is None
-        assert group.get_oldest_event() is None
 
-    def test_get_oldest_latest_events(self):
-        group = self.create_group()
-        for i in range(0, 3):
-            self.create_event(
-                event_id=six.text_type(i),
-                group=group,
-                datetime=datetime(2013, 8, 13, 3, 8, i),
-            )
-
-        assert group.get_latest_event().event_id == '2'
-        assert group.get_oldest_event().event_id == '0'
+    def test_get_latest_event(self):
+        self.store_event(
+            data={
+                'event_id': 'a' * 32,
+                'fingerprint': ['group-1'],
+                'timestamp': self.two_min_ago,
+            },
+            project_id=self.project.id,
+        )
+        self.store_event(
+            data={
+                'event_id': 'b' * 32,
+                'fingerprint': ['group-1'],
+                'timestamp': self.min_ago,
+            },
+            project_id=self.project.id,
+        )
 
-    def test_get_oldest_latest_identical_timestamps(self):
-        group = self.create_group()
-        for i in range(0, 3):
-            self.create_event(
-                event_id=six.text_type(i),
-                group=group,
-                datetime=datetime(2013, 8, 13, 3, 8, 50),
-            )
+        group = Group.objects.first()
 
-        assert group.get_latest_event().event_id == '2'
-        assert group.get_oldest_event().event_id == '0'
+        assert group.get_latest_event().event_id == 'b' * 32
 
-    def test_get_oldest_latest_almost_identical_timestamps(self):
-        group = self.create_group()
-        self.create_event(
-            event_id='0',
-            group=group,
-            datetime=datetime(2013, 8, 13, 3, 8, 0),  # earliest
-        )
-        for i in range(1, 3):
-            self.create_event(
-                event_id=six.text_type(i),
-                group=group,
-                datetime=datetime(2013, 8, 13, 3, 8, 30),  # all in the middle
+    def test_get_latest_identical_timestamps(self):
+        events = []
+        for i in 'abc':
+            event = self.store_event(
+                data={
+                    'event_id': i * 32,
+                    'fingerprint': ['group-1'],
+                    'timestamp': self.min_ago,
+                },
+                project_id=self.project.id,
             )
-        self.create_event(
-            event_id='3',
-            group=group,
-            datetime=datetime(2013, 8, 13, 3, 8, 59),  # latest
+            events.append(event)
+
+        assert events[0].group.get_latest_event().event_id == 'c' * 32
+
+    def test_get_latest_almost_identical_timestamps(self):
+        self.store_event(
+            data={
+                'event_id': 'a' * 32,
+                'fingerprint': ['group-1'],
+                'timestamp': self.two_sec_ago,
```

Comment:
I'd turn these into 60 seconds ago and 61 seconds ago to avoid test flakiness.
When you query snuba, it rounds the last timestamp to one second ago, which means you may miss the event added at line 95 one in a while.